cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/monitor/RuntimeMonitor.java

```diff
@@ -111,15 +110,22 @@ public RuntimeMonitor(ProgramRunId programRunId, CConfiguration cConf, RuntimeMo
     this.programFinishTime = -1L;
     this.lastProgramStateMessages = new LinkedList<>();
     this.requestKeyToLocalTopic = createTopicConfigs(cConf);
+    this.metricScheduledService = metricScheduledService;
   }
 
   @Override
   protected ScheduledExecutorService executor() {
     return scheduledExecutorService;
   }
 
+  @Override
+  protected void doStartUp() {
+    metricScheduledService.startAndWait();
```

Comment:
If the `startAndWait` throw exception, you should just log an error and keep going since failure to emit metric shouldn't affect the monitoring.